libbpfgo.go

```diff
@@ -1038,3 +1039,143 @@ func (pb *PerfBuffer) poll() error {
 		}
 	}
 }
+
+type TcAttachPoint uint32
+
+const (
+	BPFTcIngress       TcAttachPoint = C.BPF_TC_INGRESS
+	BPFTcEgress        TcAttachPoint = C.BPF_TC_EGRESS
+	BPFTcIngressEgress TcAttachPoint = C.BPF_TC_INGRESS | C.BPF_TC_EGRESS
+	BPFTcCustom        TcAttachPoint = C.BPF_TC_CUSTOM
+)
+
+type TcFlags uint32
+
+const (
+	BpfTcFReplace TcFlags = C.BPF_TC_F_REPLACE
+)
+
+type TcHook struct {
+	hook *C.struct_bpf_tc_hook
+}
+
+type TcOpts struct {
+	ProgFd   int
+	Flags    TcFlags
+	ProgId   uint
+	Handle   uint
+	Priority uint
+}
+
+func tcOptsToC(tcOpts *TcOpts) *C.struct_bpf_tc_opts {
+	if tcOpts == nil {
+		return nil
+	}
+	opts := C.struct_bpf_tc_opts{}
+	opts.sz = C.sizeof_struct_bpf_tc_opts
+	opts.prog_fd = C.int(tcOpts.ProgFd)
+	opts.flags = C.uint(tcOpts.Flags)
+	opts.prog_id = C.uint(tcOpts.ProgId)
+	opts.handle = C.uint(tcOpts.Handle)
+	opts.priority = C.uint(tcOpts.Priority)
+
+	return &opts
+}
+
+func tcOptsFromC(tcOpts *TcOpts, opts *C.struct_bpf_tc_opts) {
+	if opts == nil {
+		return
+	}
+	tcOpts.ProgFd = int(opts.prog_fd)
+	tcOpts.Flags = TcFlags(opts.flags)
+	tcOpts.ProgId = uint(opts.prog_id)
+	tcOpts.Handle = uint(opts.handle)
+	tcOpts.Priority = uint(opts.priority)
+}
+
+func (m *Module) TcHookInit() *TcHook {
+	hook := C.struct_bpf_tc_hook{}
+	hook.sz = C.sizeof_struct_bpf_tc_hook
+
+	return &TcHook{
+		hook: &hook,
+	}
+}
+
+func (hook *TcHook) SetInterfaceByIndex(ifaceIdx int) {
+	hook.hook.ifindex = C.int(ifaceIdx)
+}
+
+func (hook *TcHook) SetInterfaceByName(ifaceName string) error {
+	iface, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return err
+	}
+	hook.hook.ifindex = C.int(iface.Index)
+
+	return nil
+}
+
+func (hook *TcHook) GetInterfaceIndex() int {
```

Comment:
What would users need this for?

Reply:
Not sure that this would be needed, but I thought it may be good to have. I don't mind removing it though.